packages/create-llama/templates/components/workflows/typescript/artifacts/code-workflow.ts

```diff
@@ -0,0 +1,352 @@
+import { extractLastArtifact } from "@llamaindex/server";
+import { ChatMemoryBuffer, LLM, PromptTemplate, Settings } from "llamaindex";
+
+import {
+  agentStreamEvent,
+  createStatefulMiddleware,
+  createWorkflow,
+  startAgentEvent,
+  stopAgentEvent,
+  workflowEvent,
+} from "@llamaindex/workflow";
+
+import { z } from "zod";
+
+export const RequirementSchema = z.object({
+  next_step: z.enum(["answering", "coding"]),
+  language: z.string().nullable().optional(),
+  file_name: z.string().nullable().optional(),
+  requirement: z.string(),
+});
+
+export type Requirement = z.infer<typeof RequirementSchema>;
+
+const planEvent = workflowEvent<{
+  userInput: string;
+  context?: string | undefined;
+}>();
+
+const generateArtifactEvent = workflowEvent<{
+  requirement: Requirement;
+}>();
+
+const synthesizeAnswerEvent = workflowEvent<object>();
+
+const uiEvent = workflowEvent<{
```

Comment:
that problably won't look good using the ui generator - how about adding a zod schema with descriptions? (same for doc workflow)